oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedStrategy.java

```diff
@@ -276,21 +276,19 @@ public PipelinedStrategy(MongoDocumentStore documentStore,
         );
         this.mongoDocQueueSize = mongoDocQueueReservedMemoryMB / mongoDocBatchMaxSizeMB;
 
-        // Transform threads <-> merge-sort
+        // Derived values for transform <-> sort-save
+        int nseWorkingMemoryMB = readNSEBuffersReservedMemory();
         this.nseBuffersCount = 1 + numberOfTransformThreads;
-
-        long nseBuffersReservedMemoryBytes = nseBuffersReservedMemoryMB * FileUtils.ONE_MB;
+        long nseWorkingMemoryBytes = nseWorkingMemoryMB * FileUtils.ONE_MB;
```

Comment:
FileUtils.ONE_MB is a "long", so this is fine...

An option would be to cast nseWorkingMemoryMB to long here, so this is very clear.

Reply:
ok, I added the explicit cast.